ZenPacks/zenoss/ZenPackLib/lib/spec/RRDThresholdSpec.py

```diff
@@ -74,25 +74,29 @@ def create(self, templatespec, template):
 
         threshold_types = dict((y, x) for x, y in template.getThresholdClasses())
         type_ = threshold_types.get(self.type_)
-        if not type_:
-            raise ValueError("'%s' is an invalid threshold type. Valid types: %s" %
-                             (self.type_, ', '.join(threshold_types)))
 
-        threshold = template.manage_addRRDThreshold(self.name, self.type_)
-        self.speclog.debug("adding threshold")
+        if type_:
+            threshold = template.manage_addRRDThreshold(self.name, self.type_)
+            self.speclog.debug("adding threshold")
 
-        if self.dsnames is not None:
-            threshold.dsnames = self.dsnames
-        if self.eventClass is not None:
-            threshold.eventClass = self.eventClass
-        if self.severity is not None:
-            threshold.severity = int(self.severity)
-        if self.enabled is not None:
-            threshold.enabled = self.enabled
-        if self.extra_params:
-            for param, value in self.extra_params.iteritems():
-                if param in [x['id'] for x in threshold._properties]:
-                    setattr(threshold, param, value)
-                else:
-                    raise ValueError("%s is not a valid property for threshold of type %s" % (param, type_))
+            if self.dsnames is not None:
+                threshold.dsnames = self.dsnames
+            if self.eventClass is not None:
+                threshold.eventClass = self.eventClass
+            if self.severity is not None:
+                threshold.severity = int(self.severity)
+            if self.enabled is not None:
+                threshold.enabled = self.enabled
+            if self.extra_params:
+                for param, value in self.extra_params.iteritems():
+                    if param in [x['id'] for x in threshold._properties]:
+                        setattr(threshold, param, value)
+                    else:
+                        raise ValueError("%s is not a valid property for threshold of type %s" % (param, type_))
+        else:
```

Comment:
Style-wise, I'd rather see this else section moved up right under `type_ = threshold_types.get(self.type_)` as `if not type_:` and just return after logging if _optional_ was True. That way the normal-path logic doesn't have to be indented under an if.